pageserver/src/tenant/timeline.rs

```diff
@@ -854,6 +855,19 @@ impl Timeline {
         res
     }
 
+    pub(crate) async fn get_vectored_alternative(
```

Comment:
For the non-poc version of this, I think we should try and merge this with `get_vectored`. The diff between this method and the `get_vectored`. We'll have to remove the limit on the number of keys while somehow preserving a soft bound on the total amount of memory used.

Reply:
Sounds good :)